feat: Badge update

y-lohse opened the pull request:
**This is a breaking change** but the only place where `Badge` is still used is in a part of Drive that I'm changing as part of the work I'm currently doing, so it shouldn't actually break anything.

We used to have a Badge component which we didn't use much. The new spec (https://app.zeplin.io/project/584a6f4e838090371f59518f/screen/5ebd94bf1fe37a47c8638b3e) doesn't match the old components too well, but is quite in line with MUI, so this new version relies on MUI's Badge. 

Docs and use case preview: https://yannick-lohse.fr/cozy-ui/react/#!/Badge

Crash-- commented:
- Commit should have breaking change in it to create new version
- We should not only rely on our internal apps, we have external developers using UI. I don't buy the " he only place where Badge is still used is in a part of Drive". 
- Since we're not deprecated our old Badge component and replacing it without any notice, we should at least add a few lines in the commit to explain the breaking change and how to fix it.

ptbrowne commented on react/Badge/Readme.md:
```diff
@@ -1,55 +1,96 @@
-      <ButtonAction label='Normal' rightIcon='openwith' compact />
-    </Badge>
-  </p>
+### Qualifier badge
```
I wonder if variant="info" would not be better than variant="qualifier". I find "qualifier" a bit odd to be honest.

Also, I wonder if we should not have an InfoBadge component instead of choosing a variant of the badge component.

ptbrowne reviewed (commented):
I am unsure about having the ghost and qualifier variants as the API of the Badge.

* It makes our Badge different from the MUI badge
* I think the ghost variant seems very semantically close to the Drive app, and not to a general component.

I propose to separate the two badges into their own components, that use Badge as the underlying component.

What do you think ?

y-lohse commented:
@Crash-- Thanks! I'm not sure. where the "breaking change" mention went, pretty sure I added it at some point, but maybe I rewrote the commit later on. I'll add the migration step in the message 👍  

> I propose to separate the two badges into their own components, that use Badge as the underlying component.

I'll give that a try! I also found that the ghost badge was oddly specific but didn't think of splitting them 👍

y-lohse commented on react/Badge/index.jsx:
```diff
@@ -1,48 +1,165 @@
+    vertical: PropTypes.oneOf('bottom', 'top')
+  }),
+  size: PropTypes.oneOf('small', 'medium', 'large'),
+  showZero: PropTypes.bool
```
I assume you meant the values for `variant`, I added them.

y-lohse commented:
I've separated the `qualifier` and `ghost` variants into their own component. GhostBadge is there with an example but I didn't add it to the docs because it still seems very specific, but let me know what you think.

There's still the naming to decide as mentioned by @ptbrowne. I personally like `QualifierBadge`, I think it's clearer than `Info` — it's literally a badge that adds a qualification to a document.`Info` is less specific imo, but could still work.
I'm less happy with `Ghost`. Since it's so tied to the concept of files at the moment, we could name it `GhostFileBadge`?

JF-Cozy commented:
Why not `InfoBadge` ? Or `ExtraInfo` or `ExtraInfoBadge` ? I agree with @ptbrowne about `qualifier`, besides the word `information` is used in the documentation. `info` is clearer to me.

For `Ghost`, just for information, what are the different possible cases ? Is it necessarily from another Cozy instance ? What are the differences with shared files ?

y-lohse commented:
> Why not InfoBadge ? Or ExtraInfo or ExtraInfoBadge ? I agree with @ptbrowne about qualifier, besides the word information is used in the documentation. info sounds clearer to me.

Alright, I'll change it to `Infos` if  it sounds clearer :)

> For Ghost, just for information, what are the different possible cases ? Is it necessarily from another Cozy instance ? What are the differences with shared files ?

*At the moment* the only use case is the one described in the docs — in the next stack version, when Bob shares a file with Alice, a shortcut file will appear in Alice's cozy. The shortcut points to the sharing preview. Once the sharing has been accepted, the shortcut file is replaced by the actual shared file.
Emphasis on "at the moment" — we could have notes previews and bank accounts previews etc in the future? But nothing planned so far, so if you think `ghost` is too generic I don't mind using a more specific name.

y-lohse commented:
Ok, badges have been renamed, we should be good to go!

ptbrowne reviewed (approved):
Thanks !
